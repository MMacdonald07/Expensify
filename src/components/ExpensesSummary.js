import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import expensesSum from '../selectors/expenses-total';
import numeral from 'numeral';
import 'numeral/locales/en-gb';

numeral.locale('en-gb');

export const ExpensesSummary = (props) => {
	const expenseCount = props.expenses.length;
	const expenseTotal = expensesSum(props.expenses);
	const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
	return (
		<div className='page-header'>
			<div className='content-container'>
				<h1 className='page-header__title'>
					Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{numeral(expenseTotal).format('$0,0.00')}</span>
				</h1>
				<div className='page-header__actions'>
					<Link className='button' to='/create'>
						Add Expense
					</Link>
				</div>
			</div>
		</div>
	);
};

const mapStateToProps = (state) => {
	return {
		expenses: selectExpenses(state.expenses, state.filters)
	};
};

export default connect(mapStateToProps)(ExpensesSummary);
