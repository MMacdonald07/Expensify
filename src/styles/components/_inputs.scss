.text-input {
	border: 1px solid #cacccd;
	height: 50px;
	font-size: $font-size-large;
	font-weight: 300;
	padding: $s-size;
}

.select {
	@extend .text-input;
	cursor: pointer;
	-webkit-appearance: none;
	-moz-appearance: none;
	background: transparent;
	background-image: url("data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 10l5 5 5-5z'/><path d='M0 0h24v24H0z' fill='none'/></svg>");
	background-repeat: no-repeat;
	background-position-x: 100%;
	background-position-y: 10px;
}

.textarea {
	@extend .text-input;
	height: 10rem;
}
