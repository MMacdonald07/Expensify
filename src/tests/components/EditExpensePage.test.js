import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let startEditExpenseSpy, startRemoveExpenseSpy, historySpy, wrapper;

beforeEach(() => {
	startEditExpenseSpy = jest.fn();
	startRemoveExpenseSpy = jest.fn();
	historySpy = { push: jest.fn() };
	wrapper = shallow(
		<EditExpensePage
			startEditExpense={startEditExpenseSpy}
			history={historySpy}
			startRemoveExpense={startRemoveExpenseSpy}
			expense={expenses[2]}
		/>
	);
});

test('Should render EditExpensePage correctly', () => {
	expect(wrapper).toMatchSnapshot();
});

test('Should handle onSubmit', () => {
	wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2]);
	expect(historySpy.push).toHaveBeenLastCalledWith('/');
	expect(startEditExpenseSpy).toHaveBeenLastCalledWith(expenses[2].id, expenses[2]);
});

test('Should handle removeExpense', () => {
	wrapper.find('button').simulate('click');
	expect(historySpy.push).toHaveBeenLastCalledWith('/');
	expect(startRemoveExpenseSpy).toHaveBeenLastCalledWith({ id: expenses[2].id });
});
