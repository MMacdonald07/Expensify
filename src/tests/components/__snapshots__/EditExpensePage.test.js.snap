// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render EditExpensePage correctly 1`] = `
<div>
  <div
    class="page-header"
  >
    <div
      class="content-container"
    >
      <h1
        className="page-header__title"
      >
        Edit Expense
      </h1>
    </div>
  </div>
  <div
    class="content-container"
  >
    <ExpenseForm
      expense={
        Object {
          "amount": 45,
          "createdAt": 345600000,
          "description": "Credit card",
          "id": "3",
          "note": "",
        }
      }
      onSubmit={[Function]}
    />
    <button
      className="button button--secondary"
      onClick={[Function]}
    >
      Remove Expense
    </button>
  </div>
</div>
`;
